model/task_lifecycle.go

```diff
@@ -671,6 +673,118 @@ func doBisectStepback(ctx context.Context, t *task.Task) error {
+		if generated.Status != evergreen.TaskFailed {
+			return nil
+		}
+		lastPassingGenerated, err := generated.PreviousCompletedTask(generated.Project, []string{evergreen.TaskSucceeded})
+		if err != nil {
+			return errors.Wrap(err, "locating previous successful task")
+		}
+		// Noop if there is no previous passing task.
+		if lastPassingGenerated == nil {
+			return nil
+		}
+		if lastPassingGenerated.GeneratedBy == "" {
+			return errors.Errorf("last passing generated task '%s' must have a generator task", lastPassingGenerated.Id)
+		}
+		s = task.GeneratedStepbackInfo{
+			DisplayName:  generated.DisplayName,
+			BuildVariant: generated.BuildVariant,
+
+			LastPassingStepbackTaskId: lastPassingGenerated.GeneratedBy,
+		}
+		grip.Info(message.Fields{
+			"message":                       "starting bisect stepback on generator task",
+			"last_passing_stepback_task_id": s.LastPassingStepbackTaskId,
+			"generator_task_id":             generator.Id,
+			"generated_task_id":             generated.Id,
+			"display_name":                  generated.DisplayName,
+			"BuildVariant":                  generated.BuildVariant,
+			"gap":                           generated.RevisionOrderNumber - lastPassingGenerated.RevisionOrderNumber,
+			"project_id":                    generated.Project,
+		})
+	}
+
+	// The previous iteration is the task we are currently processing.
+	s.PreviousStepbackTaskId = generator.Id
+
+	// Depending on the task status, we want to update the
+	// last failing or last passing task.
+	if generated.Status == evergreen.TaskSucceeded {
+		s.LastPassingStepbackTaskId = generator.Id
+	} else if generated.Status == evergreen.TaskFailed {
+		s.LastFailingStepbackTaskId = generator.Id
+	} else {
+		return errors.Errorf("stopping task stepback due to status '%s'", generated.Status)
+	}
+
+	// The midway task is our next stepback target.
+	nextTask, err := task.FindMidwayTaskFromIds(s.LastFailingStepbackTaskId, s.LastPassingStepbackTaskId)
+	if err != nil {
+		return errors.Wrapf(err, "finding midway task between tasks '%s' and '%s'", s.LastFailingStepbackTaskId, s.LastPassingStepbackTaskId)
+	}
+	if nextTask == nil {
+		return errors.Errorf("midway task could not be found for tasks '%s' '%s'", s.LastFailingStepbackTaskId, s.LastPassingStepbackTaskId)
+	}
+	// If our next task is last passing Id, we have finished stepback.
+	if nextTask.Id == s.LastPassingStepbackTaskId {
+		return nil
+	}
+	s.NextStepbackTaskId = nextTask.Id
+	// Store our next task to our current task.
+	if err := task.SetGeneraterStepbackInfo(generator.Id, s); err != nil {
+		return errors.Wrapf(err, "could not set generated stepback task id for stepback task '%s'", generator.Id)
+	}
+	// This is only for UI purposes. The generated task needs these fields populated
+	// to show stepback options in the UI.
+	if err := task.SetLastAndPreviousStepbackIds(generated.Id, task.StepbackInfo{
+		LastFailingStepbackTaskId: "stored-in-generator",
```

Comment:
Can we just populate the fields in `s` here? Seems less confusing than having this new `"stored-in-generator"` value